Recurly/BaseClient.cs

```diff
@@ -73,8 +108,26 @@ public class BaseClient {
                 request.AddJsonBody(body);
             }
 
-            var resp = RestClient.Execute<T>(request);
-            this.ProcessResponse(resp);
+            return request;
+        }
+
+        private void HandleResponse(IRestResponse resp) {
+            if (resp.Headers.Any(t => t.Name == "Recurly-Deprecated"))
+            {
+                var headers = resp.Headers.ToList();
+                var deprecated = headers
+                    .Find(x => x.Name == "Recurly-Deprecated")
+                    .Value.ToString();
+                var sunset = headers
+                    .Find(x => x.Name == "Recurly-Sunset-Date")
+                    .Value.ToString();
+
+                if (deprecated.ToUpper() == "TRUE")
+                {
+                    Debug.WriteLine($"[recurly-client-net] WARNING: Your current API version \"${ApiVersion}\" is deprecated and will be sunset on ${sunset}");
```

Comment:
I notice that with the security warning about using `_SetApiUrl`, we use `Console.WriteLine`. Do you think that people will miss this if it's only in the Debug log? Or perhaps this isn't important enough to warrant a Console write?

Reply:
I'm not sure yet. I haven't seen any document describing how to handle this kind of logging.

Reply:
I asked a question about Debug.WriteLine vs Console.WriteLine on another slack team I'm on. One person believes that Debug.WriteLine only shows up in the IDE. So if people set up their integration and then don't make any changes, they are less likely to see this in their logs. However, they might not be logging Console.WriteLine calls either, so no matter what we run the risk that no one will see this. I think it might make sense to switch to Console.WriteLine, but I don't think that will guarantee that anyone will see this.

Reply:
Also, I can't find it in the documentation, but according to several sources I saw when I Googled it, Debug.WriteLine doesn't even get compiled into a Release build. I think we should just go ahead and change it to Console.WriteLine so it will print to stdout even in a Release build. Hopefully our users are logging calls to stdout, but that's beyond our control.

Reply:
Someone recommended logging with `ILogger` https://docs.microsoft.com/en-us/aspnet/core/fundamentals/logging/?view=aspnetcore-2.2